src/charts/stacked-bar.js

```diff
@@ -0,0 +1,1080 @@
+                return keyLabel;
+            }
+            keyLabel = _x;
+
+            return this;
+        };
+
+        /**
+         * Gets or Sets the margin of the chart
+         * @param  {Object} _x Margin object to get/set
+         * @return { margin | module} Current margin or Area Chart module to chain calls
+         * @public
+         */
+        exports.margin = function(_x) {
+           
+            if (!arguments.length) {
+                return margin;
+            }
+            margin = _x;
+
+            return this;
+        };
+
+        /**
+         * Gets or Sets the minimum width of the graph in order to show the tooltip
+         * NOTE: This could also depend on the aspect ratio
+         *
+         * @param  {Object} _x Margin object to get/set
+         * @return { tooltipThreshold | module} Current tooltipThreshold or Area Chart module to chain calls
+         * @public
+         */
+        exports.tooltipThreshold = function(_x) {
+            if (!arguments.length) {
+                return tooltipThreshold;
+            }
+            tooltipThreshold = _x;
+
+            return this;
+        };
+
+        /**
+         * Gets or Sets the valueLabel of the chart
+         * @param  {Number} _x Desired valueLabel for the graph
+         * @return { valueLabel | module} Current valueLabel or Chart module to chain calls
+         * @public
+         */
+        exports.valueLabel = function(_x) {
+            if (!arguments.length) {
+                return valueLabel;
+            }
+            valueLabel = _x;
+
+            return this;
+        };
+
+        /**
+         * Gets or Sets the number of verticalTicks of the yAxis on the chart
+         * @param  {Number} _x Desired verticalTicks
+         * @return { verticalTicks | module} Current verticalTicks or Chart module to chain calls
+         * @public
+         */
+        exports.verticalTicks = function(_x) {
+            if (!arguments.length) {
+                return verticalTicks;
+            }
+            verticalTicks = _x;
+
+            return this;
+        };
+
+        /**
+         * Gets or Sets the width of the chart
+         * @param  {Number} _x Desired width for the graph
+         * @return { width | module} Current width or Area Chart module to chain calls
+         * @public
+         */
+        exports.width = function(_x) {
+            if (!arguments.length) {
+                return width;
+            }
+            if (aspectRatio) {
+                height = Math.ceil(_x * aspectRatio);
+            }
+            width = _x;
+
+            return this;
+        };
+         /**
+         * Gets or Sets the horizontal direction of the chart
+         * @param  {number} _x Desired horizontal direction for the graph
+         * @return { horizontal | module} Current horizontal direction or Bar Chart module to chain calls
+         * @public
+         */
+        exports.horizontal = function(_x) {
```

Comment:
I prefer `.direction` make it more scalable,may add more feature in future,change it 's appearance without change code and can be configured by data input.